src/datamodel.py

```diff
@@ -64,13 +64,34 @@ def maybe_create_with_kind(owner, repo, kind):
   @staticmethod
   @ndb.tasklet
   def versions_for_key_async(key):
-    versions = yield Version.query(ancestor=key).fetch_async(keys_only=True)
+    version_cache = yield VersionCache.get_by_id_async('versions', parent=key)
+    versions = [] if version_cache is None else version_cache.versions
```

Comment:
perl syntax considered harmful. At least by me. This is a complete nitpick so feel free to ignore but I find this hella unreadable.